middle_end/flambda2/simplify/simplify_let_cont_expr.ml

```diff
@@ -794,18 +837,19 @@ let prepare_to_rebuild_one_recursive_let_cont_handler cont params
   rebuild_handler uacc
     ~after_rebuild:
       (after_one_recursive_let_cont_handler_rebuilt cont ~original_cont_scope
-         ~name_occurrences_subsequent_exprs ~after_rebuild)
+         ~name_occurrences_subsequent_exprs ~after_rebuild ~cont_uses_in_body)
 
 let after_downwards_traversal_of_one_recursive_let_cont_handler cont
     unboxing_decisions ~down_to_up params ~original_cont_scope dacc
     ~rebuild:rebuild_handler =
   let dacc = DA.map_data_flow dacc ~f:(Data_flow.exit_continuation cont) in
+  let cont_uses_in_body =
+    CUE.get_continuation_uses (DA.continuation_uses_env dacc) cont
```

Comment:
I'm not clear on why this is correct: naively, since we are after the downwards traversal of the handler, that means that we have also done the downwards pass on the body, and thus the `dacc` continuation uses env should contain all uses of the continuation. It might be correct **if** the continuation uses env is cleared after the downwards traversal pass on the body, but I haven't found the code that's doing that.